### DISCUSSION TITLE
How to access the timer object from Python

### DISCUSSION THREAD

**Discussion by sriver** [writer/writer-framework#154] [Q&A] [answered]
Hi. Would like to activate and deactivate one or more timers based on UI actions. Any idea how to access the timer object from code (assuming "Active" is the correct parameters controlling the execution of the timer)?

**Comment by ramedina86** [comment, marked as answer]
Hi, in Streamsync you don't access components directly from Python but rather set application state, which is then propagated to the frontend.

To achieve what you need you'll need to define a state element:

```py
import streamsync as ss

initial_state = ss.init_state({
    "is_timer_active": "yes",
})

def activate_timer(state):
    state["is_timer_active"] = "yes"

def deactivate_timer(state):
    state["is_timer_active"] = "no"
```

Then, in the Active property of the _Timer_, use that state element with `@{is_timer_active}`, rather than hardcoding "yes" or "no".

**Reply by sriver** [reply]
Thanks. Should have thought about that.
